app/models/hydrus/generic_object.rb

```diff
@@ -294,17 +294,17 @@ def self.stanford_terms_of_use
   # Registers an object in Dor, and returns it.
   def self.register_dor_object(*args)
     params = self.dor_registration_params(*args)
-    Dor::RegistrationService.register_object(params)
+    Dor::Services::Client.objects.register(params: params)
   end
 
   # Returns a hash of info needed to register a Dor object.
   def self.dor_registration_params(user_string, obj_typ, apo_pid)
     {
       object_type: obj_typ,
       admin_policy: apo_pid,
-      source_id: { 'Hydrus' => "#{obj_typ}-#{user_string}-#{HyTime.now_datetime_full}" },
+      source_id: "Hydrus:#{obj_typ}-#{user_string}-#{HyTime.now_datetime_full}",
```

Comment:
The param changes on lines 305 and 307 are due to the switch above on line 297, which behind the scenes now uses `Dor::RegistrationService#create_from_request` whereas it used to use `Dor::RegistrationService#register_object`. And these methods have different expectations around the args they receive.